portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -106,9 +106,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<BlogMessage> BlogMessages = getBlogsFromDatastore();
 
     // Create BlogHashMap Object and put BlogMessages in the map.
+    
+    List<BlogMessage> BlogMessagesReplies = new ArrayList<BlogMessage>();
+    for (BlogMessage message : BlogMessages) {
+      if (message.getParentID() != 0) {
```

Comment:
Consider checking the opposite condition first like:

if (message.getParentID() == 0) {
  break;
}
// Focus on your core logic.